confd/conn.go

```diff
@@ -153,6 +154,9 @@ func (c *Conn) Connect() (err error) {
 
 // Close the confd connection
 func (c *Conn) Close() (err error) {
+	if c == nil {
+		return errors.New("connection is nil")
```

Comment:
Please use a "sentinel" error, e.g. `NoConnection = errors.New("connection is nil")`